include/boost/url/impl/url.ipp

```diff
@@ -2271,63 +2274,64 @@ void
 url::
 build_tab() noexcept
```

Comment:
this should follow the fields_view_base interface of `write_table`: 
https://github.com/CPPAlliance/http_proto/blob/3168cbadff093a837f45a4414656951d154f2486/include/boost/http_proto/impl/fields_view_base.ipp#L263

Reply:
Oh... OK. I attempted to replicate the logic in `parser::parse_fields`.

Reply:
I'm not sure I know what to do here. 

1) Should we keep `build_tab` and make it call some kind of `write_table` functions somewhere else? Where? What would `buf_len_` represent in that case?
2) Or should we just reuse the logic from `write_table` in `build_tab`? If so, I'm assuming the equivalent of `fields_view_base::buf_len_` would be `url::cap_`, but the initial condition `buf_len_ > 0` (or `cap_ > 0`) will always be true, because we allocate before writing the table in the `url`.

Reply:
What I'm saying is that having the function `url_view::write_table( void* dest )` seems like it would be a good primitive, based on experience in HTTP fields. However, unlike fields we have two tables. Yes `buf_len_` in fields is the same as `cap_` in URL (maybe I will rename it heh).

I will leave the decision on implementation interface up to you.